### DISCUSSION TITLE
Comprehensive Backup and Restore for Cinnamon Settings, Including Custom Applets

### DISCUSSION THREAD

**Discussion by davidjuniorsaylor** [linuxmint/discussions#206] [Ideas]
I am an enthusiastic user of the Cinnamon desktop environment and appreciate the hard work the development team has put into making it a fantastic desktop environment. However, I have an idea that could enhance the user experience even further.

When it comes to maintaining and upgrading my system, I often find myself having to back up and restore Cinnamon settings. While the current backup feature is helpful, it does not encompass custom applets and their settings. This presents challenges because restoring the desktop environment settings might not include the settings for custom applets like "Cinnamenu."

I have an idea: it would be great to explore the feasibility of implementing a more comprehensive backup and restore mechanism for Cinnamon settings, which would also include custom applets. Such an enhancement would not only make Cinnamon more user-friendly but also improve the overall user experience.

I understand that this may be a complex feature to implement, but I'm willing to assist in any way I can, be it providing additional information, participating in testing, or contributing to the development effort.

**Comment by rcalixte** [comment]
The configuration files for Spices are stored under `~/.config/cinnamon/spices` and you can use `dconf` to dump and load settings such as key bindings (I do this myself). The downloaded Spices are also stored in `~/.local/share/cinnamon`.

While a GUI wouldn't be the worst thing, you can accomplish this in a few lines of a shell script easily enough until such a thing exists.

**Reply by anaximeno** [reply]
This could be added to the backup tool mint comes with installed by default, do you know if it is a part of a repository here?

**Reply by rcalixte** [reply]
I imagine you're referring to [Timeshift](https://github.com/linuxmint/timeshift). It includes a configuration for home directories so that should include the requested functionality already. That includes the binary form of `dconf`'s settings stored at `~/.config/dconf/user`. Similar to the shell script, you would need to know which specific directories/files to restore or migrate.

**Reply by anaximeno** [reply]
No not timeshift, there's another, look in the image below (the text is in portuguese)

![Captura de ecrã de 2023-11-13 02-31-39](https://github.com/linuxmint/discussions/assets/63073056/9d35f190-2539-43f5-90f6-c20b319d92c6)

The english name for the application would be 'Security Backup Tool' or something like that. It comes installed by default on mint.

**Reply by anaximeno** [reply]
I've found the repo, it's this one [mintbackup](https://github.com/linuxmint/mintbackup)

**Reply by rcalixte** [reply]
Ah nice. All of the settings and spices are stored in the user's home directory so this should work.
